python-avd/pyavd/_eos_designs/structured_config/base/utils.py

```diff
@@ -51,25 +66,13 @@ def _build_source_interfaces(
             # Check for duplicate VRF
             # inband_mgmt_vrf returns None in case of VRF "default", but here we want the "default" VRF name to have proper duplicate detection.
             inband_mgmt_vrf = self.shared_utils.inband_mgmt_vrf or "default"
-            if [source_interface for source_interface in source_interfaces if source_interface.get("vrf", "default") == inband_mgmt_vrf]:
+            if [source_interface for source_interface in source_interfaces if source_interface.vrf or inband_mgmt_vrf == "default"]:
                 msg = f"Unable to configure multiple {error_context} source-interfaces for the same VRF '{inband_mgmt_vrf}'."
                 raise AristaAvdError(msg)
 
-            source_interface = {"name": self.shared_utils.inband_mgmt_interface}
-            if self.shared_utils.inband_mgmt_vrf not in [None, "default"]:
-                source_interface["vrf"] = self.shared_utils.inband_mgmt_vrf
-            source_interfaces.append(source_interface)
+            source_interfaces.append_new(
+                name=self.shared_utils.inband_mgmt_interface,
+                vrf=self.shared_utils.inband_mgmt_vrf if self.shared_utils.inband_mgmt_vrf != "default" else None,
```

Comment:
so this is valid only because if `self.shared_utils.inband_mgmt_vrf` returns `None` we also inject `None` in vrf but it is not evident when reading this I think. I am partial to rewrite this to:

```suggestion
                vrf=self.shared_utils.inband_mgmt_vrf if self.shared_utils.inband_mgmt_vrf not in [None, "default"] else None,
```

for the sake of clarity. @ClausHolbechArista thoughts?

Reply:
I am ok with the current change. None will be the result for None so hey...

Reply:
typing is unhappy

Reply:
only for basedpyright